### PR TITLE
feat: disable thorchain lending for smart contract wallets

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

The follow-up of https://github.com/shapeshift/web/pull/5716 for lending

- [ ] ~~remove quotes error shoehorning~~ cc @woodenfurniture - decided to keep the shoehorning so we can still display a quote and users can switch wallets if they want to
- [x] translations everywhere

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

## Testing

- While using a smart contract wallet, ensure it is not possible to borrow / repay
- Since this consolidates the functionality of savers / THOR swaps into a single util hook while at it, ensure smart contract checks are still working there as well

### Lending

Screenshot 2023-12-04 at 14 18 44
Screenshot 2023-12-04 at 14 18 38

### THOR Swaps

Screenshot 2023-12-04 at 14 18 30

### Savers

Screenshot 2023-12-04 at 14 19 14

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #5724** <a href="https://app.graphite.dev/github/pr/shapeshift/web/5724?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/5724?utm_source=stack-comment).

File: src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Deposit/components/Confirm.tsx [hunk 1]
```diff
@@ -686,15 +686,15 @@ export const Confirm: React.FC<ConfirmProps> = ({ accountId, onNext }) => {
         <Alert status='error' width='auto' fontSize='sm'>
           <AlertIcon />
           <Stack spacing={0}>
-            <AlertTitle>Smart contract wallets not supported</AlertTitle>
+            <AlertTitle>{translate('trade.errors.smartContractWalletNotSupported')}</AlertTitle>
             <AlertDescription lineHeight='short'>
-              The THORChain network currently does not support smart contract wallets.
+              {translate('trade.thorSmartContractWalletUnsupported')}
```
Comment (woodenfurniture) [inline src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Deposit/components/Confirm.tsx:691]: 🍻

File: src/pages/Lending/Pool/components/Borrow/BorrowInput.tsx [hunk 2]
```diff
@@ -275,6 +278,33 @@ export const BorrowInput = ({
     isEstimatedSweepFeesDataSuccess,
   ])
 
+  const { data: _isSmartContractAddress, isLoading: isAddressByteCodeLoading } = useQuery({
+    queryKey: [
+      'isSmartContractAddress',
+      {
+        userAddress: collateralAccountId
```
Comment (kaladinlight) [inline src/pages/Lending/Pool/components/Borrow/BorrowInput.tsx:285]: could memoize the `userAddress` to use throughout this `useQuery` and in `disableSmartContractDeposit`?

File: src/pages/Lending/Pool/components/Borrow/BorrowInput.tsx [hunk 3]
```diff
@@ -275,6 +278,33 @@ export const BorrowInput = ({
...
+    queryFn: () =>
+      isSmartContractAddress(
+        collateralAccountId ? fromAccountId(collateralAccountId).account.toLowerCase() : '',
+      ),
+  })
+
+  const disableSmartContractDeposit = useMemo(() => {
+    // Collateral AccountId still loading - disable confirm
+    if (!collateralAccountId) return true
+    if (!isAddress(fromAccountId(collateralAccountId).account)) return false
```
Comment (kaladinlight) [inline src/pages/Lending/Pool/components/Borrow/BorrowInput.tsx:299]: could reduce to one `if (!isAddress(userAddress)` check with the memoized `userAddress` variable. Also, shouldn't we be returning `true` if it is not a valid address? would be more of a generic disable func as it disables button for invalid or smart contract address, but seems like what we want

File: src/pages/Lending/Pool/components/Borrow/BorrowInput.tsx (hunk 2 above)
Comment (gomesalexandre) [reply src/pages/Lending/Pool/components/Borrow/BorrowInput.tsx]: adffb67841e166bac0eea41ef1fcbb3448136a67

**Review by kaladinlight — APPROVED** [review]

**Pull request merged**